tiled/server/app.py

```diff
@@ -298,8 +300,23 @@ async def unsupported_query_type_exception_handler(
         allow_credentials=True,
         allow_methods=["*"],
         allow_headers=["*"],
+        # allow_headers=["X-Requested-With", "X-Request-ID"],
+        expose_headers=["X-Tiled-Request-ID"],
     )
 
+    @app.exception_handler(Exception)
+    async def unhandled_exception_handler(
+        request: Request, exc: Exception
+    ) -> JSONResponse:
+        return await http_exception_handler(
+            request,
+            HTTPException(
```

Comment:
Just curious, were unhandled exceptions returning `500` before this? Would this get swallowed if the prometheus metrics were also being produced? I say this looking at the fact that on line 871, promethus client code is also catching Exception and raising 500.

Reply:
Oooh, I see I may have been misreading this...the 500 in prometheus code is only generating if it catches an exception in doing metrics things. 

I'll put a comment there to be more clear.

Reply:
Does [4b5aaa9](https://github.com/bluesky/tiled/pull/673/commits/4b5aaa9947d926488650ebec19da1e7fd5630bbd) seem right?